dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/newsession/NewSessionPresenter.java

```diff
@@ -289,6 +291,20 @@ public void saveSession() {
         }
     }
 
+    @NotNull
+    private String buildSessionAlert(final Session session) {
+        final String message;
+        final int index = savedSessionStore.getSessions().stream()
+                .map(SavedSession::getName)
+                .collect(GuavaCollectors.immutableList()).indexOf(session.getSessionName());
```

Comment:
This seems like you could use a filter, and the find first to implement this instead of turning it into a list and then iterating over the list to find the entry.